subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java

```diff
@@ -202,4 +202,21 @@
      */
     @Incubating
     List<Capability> getRequestedCapabilities();
+
+    /**
+     * Inherit version constraints with {@link VersionConstraint#isForSubgraph()} from the target module.
```

Comment:
:x: I have read the docs several times but I still can't understand what this means. It corresponds to your old "assimilate" method call, but the docs should rather explain what the semantics are rather than referring to an orthogonal method (`isForSubgraph`). This makes it quite confusing.

Also, does it _really_ inherit _subgraph constraints_ (all the constraints from the subgraph), or does it only integrate the constraints from the target node to itself? I understand the latter, so it should rather be something like `inheritTargetConstraints()` or `integrateTargetConstraints()` or `assimilateTargetConstraints()`.

Reply:
Agreed the documentation is not clear yet. I choose the name because I was looking for a name for this thing:

`constraints { api("org:module:1.0") { isForSubgraph() } }`

I.e. a constraint that has the `isForSubgraph()` flag set. I called that a _subgraph constraint_ and hence the name `inheritSubgraphConstraints()`. Which does not mean "inherit all constraints from the subgraph" but "inherit _subgraph_ constraints from the target". (It does also not means "inherit *all* constraints from the target", although maybe this does not matter, because all _non-subgraph_ constraints apply globally anyway).

Maybe `inheritTargetConstraints()` is a better name. (But lets discussing naming again offline before I refactor this.)